### PR TITLE
Assert a keyword-only parameter stays keyword-only, library-wide (issue 980)

### LINKED ISSUE #980: Nothing asserts a keyword-only parameter is keyword-only: 89 public callables, 132 parameters, every mutant alive
Three of `fetch`'s survivors are the keyword-only marker:

```python
    def __init__(
        self,
        client: BitcoinCoreRpcClient,
        network: str = "mainnet",
        *,                          # ReplaceBinaryOperator_Mul_Div: `/`
        verify_network: bool = True,
        signet_challenge: str | bytes | None = None,
    ) -> None:
```

`*` and `/` are two markers with opposite effects — `*` makes what follows
keyword-only, `/` makes what precedes positional-only — so the mutation
drops the keyword-only rule and adds a positional-only one in its place.
Every test still passes, in `BitcoinCoreFetcher` and in
`EsploraFetcher.__init__` (`network`, `timeout`, `transport`) alike.

This is not a `fetch` problem. Measured on `main` by walking every
`__all__` and reading `inspect.signature`, deduplicated by object
identity so a re-export is not counted twice:

- **89 public callables** take keyword-only parameters
- **132 distinct keyword-only parameters** among them
- and the recurring ones are the library's own conventions:
  `check_validity` 25 times, `network` 16, `receipt` 8, `commit` and
  `commit_hash` and `leaf_hash` 6 each, `timeout` 5

Nothing in `tests/` asserts that any of them is keyword-only. So the
mutant survives at all 89 sites, and — the part that matters more than the
mutation score — a refactor that made `check_validity` positional would
pass the whole suite.

## Why it is a decision and not a chore

The obvious objection to writing 89 tests is the right one, and it has an
answer: the property is mechanical, so one parametrized test over the
public surface covers every site at once and keeps covering new ones. The
enumeration above *is* that test, near enough — walk `__all__`, read the
signature, assert the parameter's `kind`.

That makes the question about intent rather than cost:

1. **Assert it library-wide**, one test parametrized over the public
   surface. Cheap, total, and it turns the calling convention into
   something the suite owns: `check_validity` cannot quietly become
   positional in 25 places. The cost is a test that fails whenever
   somebody *deliberately* changes a signature, which is either the point
   or an annoyance depending on the answer to this issue.
2. **Assert it where the convention carries weight** — `check_validity`
   and `network`, the two that recur — and leave the rest.
3. **Nothing.** A keyword-only marker is a source-compatibility promise
   rather than a behaviour, and `CHANGELOG.md` already treats a signature
   change as a breaking change to be written down. The mutants stay alive
   and the survivor list says why.

No preference offered. What decides it is whether btclib considers the
calling convention part of the contract the suite defends or part of the
documentation, and that answer also settles how the eleven other
profiles' `*` mutants should read.

Related: [#975](https://github.com/btclib-org/btclib/issues/975) asks the
same shape of question about the bindings delegation — a claim three files
make and no test holds. This one is a claim 89 signatures make.

### PR DESCRIPTION
**Pull request by fametrano**
## Summary

`*` in a signature is a calling-convention promise — `check_validity`,
`network`, and a dozen other names that recur — and nothing in the
suite tested it: a mutant turning `*` to `/` drops the keyword-only
rule and adds a positional-only one in its place, and every test still
passed, at every public callable that takes a keyword-only parameter.

Chose option 1 from #980: one test parametrized over the public
surface, walking `__all__` and reading `inspect.signature`.

## Changes

`tests/keyword_only_test.py` is new. `KEYWORD_ONLY` is a frozen table
— walked once from the current tree, every public callable (a class
named by its `__init__`) that takes at least one keyword-only
parameter, deduplicated by object identity so a re-export is not
counted twice. Two tests read it:

- `test_a_keyword_only_parameter_stays_keyword_only` — parametrized
  over every `(callable, parameter)` pair in the table; for each, reads
  the callable's *live* signature and asserts the parameter's `kind` is
  still `KEYWORD_ONLY`. A single mutant of one `*` fails by name rather
  than by count.
- `test_the_recorded_surface_is_the_whole_of_it` — recomputes the same
  walk against the current tree and diffs it against the frozen table,
  so a keyword-only parameter added to the public surface without a
  line here fails a test instead of running none.

The table is frozen rather than recomputed at test time on purpose:
recomputing it from the same code the test is meant to guard would read
whatever a mutation had just done and call that the answer.

Closes #980.

## Test plan

- [x] `uv run pytest -q` — 28092 passed, 100% coverage
- [x] Verified the new tests fail against the mutant `#980` names
      (`BitcoinCoreFetcher.__init__`'s `*` turned to `/`), and pass on
      `main`'s code
- [x] `uv run pre-commit run --files tests/keyword_only_test.py
      CHANGELOG.md` — all hooks pass
- [x] `sphinx-build -W --keep-going` — build succeeded

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Protect the public API’s keyword-only calling conventions with comprehensive signature coverage.

Enhancements:
- Add library-wide tests that verify every recorded public keyword-only parameter retains its keyword-only calling convention.
- Cover public classes and methods, including alternate constructors and re-exported callables, while checking that the recorded surface remains complete.

Tests:
- Add parametrized signature tests for all public keyword-only parameters and a completeness check against the exported API surface.

### CODE REVIEW

File: tests/keyword_only_test.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        "verify_network",
+        "signet_challenge",
+    ],
+    "btclib.fetch:BitcoinCoreRpcClient.__init__": [
+        "user",
+        "password",
+        "cookie_path",
+        "timeout",
+        "transport",
+    ],
```
[the remark this answers is not in the document]
Comment (fametrano) [reply tests/keyword_only_test.py:175]: Confirmed — good catch. `_live_keyword_only` widened to walk every name in a class's own `__dict__` that isn't private (`__init__` plus every alternate constructor and instance method), not only `__init__`, deduplicated by the underlying function's id so an inherited-and-not-overridden method isn't walked twice. Regenerated `KEYWORD_ONLY` from the widened walk — 90 → 182 callables, 139 → 246 sites — and verified your own example, `Bip21.parse`'s `*` → `/`, now fails both `test_a_keyword_only_parameter_stays_keyword_only[btclib.bip21:Bip21.parse:check_validity]` and the completeness test. Pushed in 9a18c7bf.

**Comment by fametrano** [comment]
Noted on the theoretical edge case (a keyword-only method on a base class not itself exported via `__all__`, which the per-class `vars(cls)` walk — rather than a full MRO walk — would miss on an exported subclass). Checked the current tree the same way you did: no such hierarchy exists today, and adding an MRO walk now would be guarding against a shape nothing in btclib has, so leaving `_class_sites` as is rather than generalizing for a case that isn't there yet. If a hierarchy like that shows up later, `test_the_recorded_surface_is_the_whole_of_it` would still catch the new keyword-only site being missing from `KEYWORD_ONLY` — it just wouldn't attribute it correctly to the base class until the walk is widened.

**Comment by fametrano** [comment]
Correction to my last comment: I was wrong that \`test_the_recorded_surface_is_the_whole_of_it\` would still catch this. I checked directly — \`vars(Sub)\` for a subclass that inherits (and does not override) a method from an unexported base class does **not** contain that method name at all:

\`\`\`python
class Base:
    @classmethod
    def parse(cls, x, *, flag=True): ...

class Sub(Base):
    pass

'parse' in vars(Sub)  # False
\`\`\`

So the walk over the exported \`Sub\` would never see \`parse\`, and since both \`KEYWORD_ONLY\` (frozen) and \`_live_keyword_only()\` (live recompute) use the same walk, they'd agree in omitting it — the same blindness this PR exists to remove, reintroduced at one more level. Not a live bug today (verified no such hierarchy exists in the current tree, as you did), but the safety net I described doesn't actually apply to this specific gap. Leaving the walk as `vars(cls)`-only for now since there's nothing to guard against yet, but wanted to correct the record rather than leave a false claim standing.

**Comment by fametrano** [comment]
pre-commit.ci run
